include/libtorrent/block_cache.hpp

```diff
@@ -337,7 +332,7 @@ namespace libtorrent
 		struct hash_value
 		{
 			std::size_t operator()(cached_piece_entry const& p) const
-			{ return std::size_t(p.storage.get()) + std::size_t(p.piece); }
+			{ return std::size_t(p.storage.get()) + static_cast<int>(p.piece); }
```

Comment:
oh, I see, the cast to ``std::size_t`` should probably be moved to encompass the ``+`` expression